ads1x15: race condition in unit test

https://gist.github.com/gohci-bot/1ba0fd321ddcb295bcd256e2adfe80fb

```
$GOPATH/src/periph.io/x/periph $ go test -race ./...  (exit:1 in 55.551s)
(...)
panic: i2ctest: unexpected Tx() (count #6) expecting i2ctest.IO{Addr:72, W:[]byte{0x1, 0x91, 0x3}, R:[]byte(nil)}

goroutine 6 [running]:
periph.io/x/periph/conn/i2c/i2ctest.errorf(0xc4200c6000, 0x11e48d6, 0x50, 0xc420047d30, 0x4, 0x4, 0x1, 0x2)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/conn/i2c/i2ctest/i2ctest.go:166 +0xe1
periph.io/x/periph/conn/i2c/i2ctest.(*Playback).Tx(0xc4200c6000, 0x1050048, 0xc4200620d0, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/conn/i2c/i2ctest/i2ctest.go:127 +0xc79
periph.io/x/periph/conn/i2c.(*Dev).Tx(0xc420062080, 0xc4200620d0, 0x3, 0x3, 0x0, 0x0, 0x0, 0xc420047f48, 0x2)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/conn/i2c/i2c.go:94 +0xb9
periph.io/x/periph/experimental/devices/ads1x15.(*Dev).executePreparedQuery(0xc420062080, 0xc4200620d0, 0x3, 0x3, 0x78bc34, 0x16e360000, 0x0, 0x0, 0x0, 0x0)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/experimental/devices/ads1x15/ads1x15.go:250 +0xda
periph.io/x/periph/experimental/devices/ads1x15.(*analogPin).Read(0xc4200620c0, 0x1, 0x0, 0x0, 0x0)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/experimental/devices/ads1x15/ads1x15.go:412 +0xbf
periph.io/x/periph/experimental/devices/ads1x15.(*analogPin).ReadContinuous.func1(0xc4200c6050, 0xc4200cc000, 0xc4200620c0, 0xc42001e180)
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/experimental/devices/ads1x15/ads1x15.go:438 +0x16b
created by periph.io/x/periph/experimental/devices/ads1x15.(*analogPin).ReadContinuous
	/Users/gohci/gohci/google_periph/src/periph.io/x/periph/experimental/devices/ads1x15/ads1x15.go:430 +0x1af
FAIL	periph.io/x/periph/experimental/devices/ads1x15	0.070s

```

--- a/experimental/devices/ads1x15/ads1x15.go
+++ b/experimental/devices/ads1x15/ads1x15.go
@@ -226,9 +226,8 @@ func (d *Dev) PinForChannel(c Channel, maxVoltage physic.ElectricPotential, f ph
 	configBytes := [2]byte{}
 	binary.BigEndian.PutUint16(configBytes[:], config)
 
-	// The wait for the ADC sample to finish is based on the sample rate plus a
-	// small offset to be sure (0.1 millisecond).
-	waitTime := time.Second/time.Duration(dataRate) + 100*time.Microsecond
+	// The wait for the ADC sample to finish is based on the sample rate.
+	waitTime := time.Second / time.Duration(dataRate)
 
 	return &analogPin{
 		adc:                d,

--- a/experimental/devices/ads1x15/ads1x15_test.go
+++ b/experimental/devices/ads1x15/ads1x15_test.go
@@ -165,6 +165,7 @@ func TestPinADC_ReadContinous(t *testing.T) {
 				R:    []byte{0x52, 0xc0},
 			},
 		},
+		DontPanic: true,
 	}
 
 	rawValues := []int32{21200, 21184}

--- a/experimental/devices/ads1x15/ads1x15_test.go
+++ b/experimental/devices/ads1x15/ads1x15_test.go
@@ -76,7 +76,7 @@ func TestPinADC_Read(t *testing.T) {
 		Ops: []i2ctest.IO{
 			{
 				Addr: 0x48,
-				W:    []byte{0x1, 0x91, 0xc3},
+				W:    []byte{0x1, 0x91, 0x3},
 				R:    []byte{},
 			},
 			{
@@ -92,7 +92,7 @@ func TestPinADC_Read(t *testing.T) {
 		t.Fatal(err)
 	}
 	// Obtain an analog pin from the ADC
-	pin, err := d.PinForChannel(Channel0Minus3, 5*physic.Volt, 1*physic.Hertz, SaveEnergy)
+	pin, err := d.PinForChannel(Channel0Minus3, 5*physic.Volt, 1*physic.Hertz, BestQuality)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -120,13 +120,12 @@ func TestPinADC_Read(t *testing.T) {
 	}
 }
 
-/* https://github.com/google/periph/issues/334
 func TestPinADC_ReadContinous(t *testing.T) {
 	b := i2ctest.Playback{
 		Ops: []i2ctest.IO{
 			{
 				Addr: 0x48,
-				W:    []byte{0x1, 0x91, 0x3},
+				W:    []byte{0x1, 0x91, 0xc3},
 				R:    []byte{},
 			},
 			{
@@ -136,7 +135,7 @@ func TestPinADC_ReadContinous(t *testing.T) {
 			},
 			{
 				Addr: 0x48,
-				W:    []byte{0x1, 0x91, 0x3},
+				W:    []byte{0x1, 0x91, 0xc3},
 				R:    []byte{},
 			},
 			{
@@ -144,10 +143,20 @@ func TestPinADC_ReadContinous(t *testing.T) {
 				W:    []byte{0x0},
 				R:    []byte{0x52, 0xc0},
 			},
-			// We add one extra exchange, as halting the polling is not instant
+			// We add 2 extra exchanges, as halting the polling is not instant
 			{
 				Addr: 0x48,
-				W:    []byte{0x1, 0x91, 0x3},
+				W:    []byte{0x1, 0x91, 0xc3},
+				R:    []byte{},
+			},
+			{
+				Addr: 0x48,
+				W:    []byte{0x0},
+				R:    []byte{0x52, 0xc0},
+			},
+			{
+				Addr: 0x48,
+				W:    []byte{0x1, 0x91, 0xc3},
 				R:    []byte{},
 			},
 			{
@@ -166,7 +175,7 @@ func TestPinADC_ReadContinous(t *testing.T) {
 		t.Fatal(err)
 	}
 	// Obtain an analog pin from the ADC
-	pin, err := d.PinForChannel(Channel0Minus3, 5*physic.Volt, 100*physic.Hertz, BestQuality)
+	pin, err := d.PinForChannel(Channel0Minus3, 5*physic.Volt, 100*physic.Hertz, SaveEnergy)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -198,4 +207,3 @@ func TestPinADC_ReadContinous(t *testing.T) {
 		t.Fatal(err)
 	}
 }
-*/

